scripts/release/release.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/sh
+
+SCRIPTS_DIR=$(dirname "$0")
+BASE_DIR="$(pwd)"
+SRC_DIR="$BASE_DIR/libelektra"
+BUILD_DIR="$SRC_DIR/build"
+FTP_DIR="$BASE_DIR/ftp"
+DOC_DIR="$BASE_DIR/docu"
+
+install_elektra() {
+	echo "Installing elektra..."
+	rm -rf $BUILD_DIR && mkdir $BUILD_DIR && cd $BUILD_DIR && cmake -DBUILD_SHARED=ON -DBUILD_FULL=ON -DBUILD_STATIC=ON ..
+	cd ..
+	cmake --build build -- -j5
+	cd $BUILD_DIR
+	make install
+	export VERSION=$(kdb get system/elektra/version/constants/KDB_VERSION)
+	export DVERSION=$VERSION-1
+}
+
+run_updates() {
+	echo "Running updates..."
+	mkdir $BASE_DIR/$VERSION
+	# regenerate dot of plugins
+	$SRC_DIR/scripts/dev/draw-all-plugins > $BASE_DIR/$VERSION/draw-all-plugins                         # TODO: rename file
+	# run link checker
+	cd $BUILD_DIR && $SRC_DIR/scripts/link-checker external-links.txt > $BASE_DIR/$VERSION/link-checker # TODO: rename file
+}
+
+update_debian_version_number() {
+	echo "Updating debian version..."
+	cd $SRC_DIR
+	dch --newversion $DVERSION "New upstream version."
+	git add debian/changelog
+	git commit -m "Debian Package $DVERSION (UNRELEASED)"
+}
+
+cleanup() {
+	echo "Running cleanup..."
+	git checkout master
+
+	# merge any left over feature branches
+	git merge --no-ff feature -m "Merge feature branches for ${VERSION} release"
+	git commit -a -m "Release preperation for ${VERSION}"
+	# cleanup
+	git clean -dfx
+
+	# make an empty builddirectory:
+	mkdir $BUILD_DIR
+}
+
+run_checks() {
+	echo "Running checks..."
+
+	# check build-server
+	kdb --version > $BASE_DIR/$VERSION/version
+
+	# Rebuild cleanly, run all tests and also check for memleaks:
+	cd "$BUILD_DIR" && $SRC_DIR/scripts/dev/configure-debian "$SRC_DIR" && make -j24 && make run_all -j24 && make run_memcheck -j24
+
+	# Check if there are really >=241 or >=131 tests
```

Comment:
@markus2330 are these numbers still relevant today, as they are always changing? Basically anything below 200 tests would seem suspicious to me (on debian buster).

@robaerd this can be adapted later, I just wanted to raise this question.

Reply:
These are already quite old numbers but to make the test too strict is also not a good idea (if we remove one plugin, we would immediately get an failure). We need some experience with the new pipeline to find out what we really want, I am currently imagining to compare the logs between the releases. But some hard checks to find crucial problems also will not harm.